{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import graphlab as gl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ec2config = gl.deploy.Ec2Config(region='us-east-1',\n",
    "                                instance_type = 'c4.xlarge',\n",
    "                                aws_access_key_id = 'AKIAIAROZHHVYJAJVGIQ',\n",
    "                                aws_secret_access_key = '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',\n",
    "                                security_group = ''\n",
    "                               )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "Exception",
     "evalue": "Unable to launch EC2 instance: 'EC2ResponseError: 400 Bad Request\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Response><Errors><Error><Code>AuthFailure</Code><Message>Not authorized for images: [ami-911c6af4]</Message></Error></Errors><RequestID>e96ce73c-b652-4fd4-a7fd-3942c84b79f4</RequestID></Response>'. Please check AWS Console to make sure any EC2 instances launched have been terminated.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mException\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-83-324e583134de>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m my_cluster = gl.deploy.ec2_cluster.create('test_cluster',\n\u001b[1;32m      2\u001b[0m                                          \u001b[0;34m's3://imageclassification/'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m                                           ec2_config = ec2config)\n\u001b[0m",
      "\u001b[0;32m/Users/galen/anaconda/envs/gl-env/lib/python2.7/site-packages/graphlab/deploy/ec2_cluster.pyc\u001b[0m in \u001b[0;36mcreate\u001b[0;34m(name, s3_path, ec2_config, num_hosts, additional_packages, idle_shutdown_timeout)\u001b[0m\n\u001b[1;32m     81\u001b[0m     \u001b[0mcluster\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mEc2Cluster\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mec2_config\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0ms3_path\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnum_hosts\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0madditional_packages\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0midle_shutdown_timeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     82\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 83\u001b[0;31m     \u001b[0mcluster\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstart\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     84\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mcluster\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     85\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/galen/anaconda/envs/gl-env/lib/python2.7/site-packages/graphlab/deploy/ec2_cluster.pyc\u001b[0m in \u001b[0;36mstart\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    231\u001b[0m                                                                        \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnum_hosts\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    232\u001b[0m                                                                        \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0madditional_packages\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 233\u001b[0;31m                                                                        \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0midle_shutdown_timeout\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    234\u001b[0m                                                                        )\n\u001b[1;32m    235\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_save\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/galen/anaconda/envs/gl-env/lib/python2.7/site-packages/graphlab/deploy/_executionenvironment.pyc\u001b[0m in \u001b[0;36m_start_commander_host\u001b[0;34m(env_name, config, s3_folder_path, num_hosts, additional_packages, idle_shutdown_timeout)\u001b[0m\n\u001b[1;32m    356\u001b[0m                                 \u001b[0mproduct_type\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_ProductType\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTuriDistributed\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    357\u001b[0m                                 \u001b[0msubnet_id\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconfig\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msubnet_id\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 358\u001b[0;31m                                 security_group_id = config.security_group_id)\n\u001b[0m\u001b[1;32m    359\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    360\u001b[0m         \u001b[0;31m# Log message explaining the additional hosts will not be launched by us.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/galen/anaconda/envs/gl-env/lib/python2.7/site-packages/graphlab/connect/aws/_ec2.pyc\u001b[0m in \u001b[0;36m_ec2_factory\u001b[0;34m(instance_type, region, availability_zone, CIDR_rule, security_group_name, tags, user_data, credentials, ami_service_parameters, num_hosts, additional_port_to_open, product_type, subnet_id, security_group_id)\u001b[0m\n\u001b[1;32m    519\u001b[0m             \u001b[0m_stop_instances\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mid\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0minstances\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m,\u001b[0m \u001b[0mregion\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    520\u001b[0m         raise Exception(\"Unable to launch EC2 instance: '%s'. Please check AWS Console to make\"\n\u001b[0;32m--> 521\u001b[0;31m                 \" sure any EC2 instances launched have been terminated.\" % e)\n\u001b[0m\u001b[1;32m    522\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    523\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0m_get_ec2_instances\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mec2_id_list\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mregion\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maws_credentials\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mException\u001b[0m: Unable to launch EC2 instance: 'EC2ResponseError: 400 Bad Request\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Response><Errors><Error><Code>AuthFailure</Code><Message>Not authorized for images: [ami-911c6af4]</Message></Error></Errors><RequestID>e96ce73c-b652-4fd4-a7fd-3942c84b79f4</RequestID></Response>'. Please check AWS Console to make sure any EC2 instances launched have been terminated."
     ]
    }
   ],
   "source": [
    "my_cluster = gl.deploy.ec2_cluster.create('test_cluster',\n",
    "                                         's3://imageclassification/',\n",
    "                                          ec2_config = ec2config)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
